### ISSUE TITLE
[typescript][enhancement]: Update typescript sample with the latest typescript definitions 

### ISSUE THREAD

**Issue by abisalehalliprasan** [intuit/oauth-jsclient#99]
Hacktoberfest is underway right now and we would love for you to contribute  🚀  . More information on #hacktoberfest is [here](https://hacktoberfest.digitalocean.com/)

[Develop-with-Intuit for Hacktoberfest](https://medium.com/intuit-engineering/develop-with-intuit-for-hacktoberfest-2020-8f29b70ff2b4) 

### Problem
Update typescript sample with the latest typescript definitions index.d.ts [file](https://github.com/intuit/oauth-jsclient/blob/typescript/index.d.ts) 

### Fix Needed 
Update [typescript sample](https://github.com/intuit/oauth-jsclient/tree/typescript/sample/typescript) with the latest typescript definitions and test the sample is working.

### Instructions to use the sample app 

In order to test the typings locally from the sample project, you'll need to link the `intuit-oauth` package with yarn or npm.

```
# Link local intuit-oauth branch with typings support to sample typescript project
yarn link
cd sample/typescript
yarn link intuit-oauth
```

### PR Guidelines 

Kindly follow the below instructions to submit your PR :  

- Fork and clone the repository (`typescript` branch).
- Run `npm install` for dependencies.
- Test all your changes.
- Run `npm test` to execute all specs.

Once ready, please submit a PR to the `typescript` branch [here](https://github.com/intuit/oauth-jsclient/tree/typescript).  
 `Note :` PR's will be merged to the `master` branch once all the testing is completed.

**Comment by 100mik** [comment]
Hey! I'd like to take a shot at this if it's fine.

**Comment by abisalehalliprasan** [comment]
@100mik : Awesome. That would be great. However, this issue/fix would be dependent on #100 ( as the sample should refer to the latest typings ). 

I suggest you can start on this story once we start getting more updates on the #100. Does that sound good to you?

**Comment by 100mik** [comment]
Yes that's perfectly fine. I don't mind helping out if needed once the current assignee has taken a shot at #100 
Cheers!
